hoomd/hpmc/ShapeMoves.h

```diff
@@ -0,0 +1,870 @@
+    {
+    public:
+    ConstantShapeMove(std::shared_ptr<SystemDefinition> sysdef,
+                      std::shared_ptr<IntegratorHPMCMono<Shape>> mc,
+                      pybind11::dict shape_params)
+        : ShapeMoveBase<Shape>(sysdef, mc), m_shape_moves({})
+        {
+        std::vector<pybind11::dict> shape_params_vector(this->m_ntypes);
+        for (auto name_and_params : shape_params)
+            {
+            std::string type_name = pybind11::cast<std::string>(name_and_params.first);
+            pybind11::dict type_params = pybind11::cast<pybind11::dict>(name_and_params.second);
+            unsigned int type_i = this->m_sysdef->getParticleData()->getTypeByName(type_name);
+            shape_params_vector[type_i] = type_params;
+            }
+        m_shape_params = shape_params_vector;
+        for (unsigned int i = 0; i < this->m_ntypes; i++)
+            {
+            typename Shape::param_type pt(m_shape_params[i]);
+            m_shape_moves.push_back(pt);
+            }
+        if (this->m_ntypes != m_shape_moves.size())
+            throw std::runtime_error("Must supply a shape move for each type");
+        for (unsigned int i = 0; i < m_shape_moves.size(); i++)
+            {
+            detail::MassProperties<Shape> mp(m_shape_moves[i]);
+            m_determinants.push_back(mp.getDetInertiaTensor());
+            }
+        }
+
+    void prepare(uint64_t timestep) { }
+
+    void update_shape(uint64_t timestep,
+                      Scalar& stepsize,
+                      const unsigned int& type_id,
+                      typename Shape::param_type& shape,
+                      hoomd::RandomGenerator& rng)
+        {
+        shape = m_shape_moves[type_id];
+        this->m_det_inertia_tensor = m_determinants[type_id];
+        }
+
+    void retreat(uint64_t timestep)
+        {
+        // move has been rejected.
+        }
+
+    pybind11::dict getShapeParams()
+        {
+        pybind11::dict shape_params;
+        for (unsigned int i = 0; i < m_shape_params.size(); i++)
+            {
+            pybind11::str type_name = this->m_sysdef->getParticleData()->getNameByType(i);
+            shape_params[type_name] = m_shape_params[i];
+            }
+        return shape_params;
+        }
+
+    void setShapeParams(pybind11::dict shape_params)
+        {
+        std::vector<pybind11::dict> shape_params_vector(m_shape_params.size());
+        for (auto name_and_params : shape_params)
+            {
+            std::string type_name = pybind11::cast<std::string>(name_and_params.first);
+            pybind11::dict type_params = pybind11::cast<pybind11::dict>(name_and_params.second);
+            unsigned int type_i = this->m_sysdef->getParticleData()->getTypeByName(type_name);
+            shape_params_vector[type_i] = type_params;
+            typename Shape::param_type pt(type_params);
+            m_shape_moves[type_i] = pt;
+            }
+        m_shape_params = shape_params_vector;
+        }
```

Comment:
These should be type specific.

Reply:
Agree. The `ConstantShapeMove` still needs some work as I've been prioritizing the other more relevant moves. I don't know how exactly it is supposed to work nor what its intended purpose is.

Reply:
Then you can remove it.

Reply:
Done